### ISSUE TITLE
Consider logging item adds/removes during evaluation

### ISSUE THREAD

**Issue by KirillOsenkov** [Microsoft/msbuild#2469]
This issue https://github.com/KirillOsenkov/MSBuildStructuredLog/issues/113 brings up a point: items added and removed during evaluation are not logged, so the log doesn't have any trace of it.

We should consider logging item removes (not sure if logging item adds is valuable because items added and not removed are logged per project).

I'm on the fence but I think triage should discuss and consider.

**Comment by rainersigwald** [comment]
Yeah, that's an interesting one. One wrinkle is that removes aren't necessarily "removed" as in added and then removed, thanks to lazy item evaluation. That collapses

```xml
<ItemGroup>
  <X Include="*.cs" />
  <X Remove="*.xaml.cs" />
</ItemGroup>
```

into basically

```xml
<ItemGroup>
  <X Include="*.cs" Exclude="*.xaml.cs" />
</ItemGroup>
```

In the OM, items are annotated with the XML element that caused them to come about. Maybe we could  add something similar to the binary log format?

**Comment by KirillOsenkov** [comment]
Good idea. I've opened a new issue to track that work:
https://github.com/Microsoft/msbuild/issues/2473
